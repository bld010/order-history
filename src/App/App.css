.App {
  text-align: center;
}

header {
  align-items: center;
  background: url('../images/checkout.jpg') no-repeat center / 100%;
  color: #FFF;
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-around;
  height: 300px;
}

h1 {
  font-size: 72px;
}



.purchase-container {
  height: auto;
  min-height: 700px;
  background:#CECECE;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}