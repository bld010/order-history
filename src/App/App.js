import React, { Component } from 'react';
import './App.css';
import PurchasesContainer from '../PurchasesContainer/PurchasesContainer';
import Form from '../Form/Form';

class App extends Component {

  constructor() {
    super();
    this.state = {
      purchases: []
    }
  }

  addNewPurchase = (e, newPurchase) => {
    e.preventDefault();
    this.setState({purchases: [...this.state.purchases, newPurchase]});
    this.postNewPurchase(newPurchase);
  }

  deletePurchase = (e, idOfPurchase) => {
    let updatedPurchases = this.state.purchases.filter( purchase => {
      return purchase.id !== idOfPurchase
    })
    this.setState({purchases: updatedPurchases});
    this.deletePurchaseFetch(idOfPurchase)
  } 

  deletePurchaseFetch = (id) => {
    fetch(`http://localhost:3001/api/v1/purchases/${id}`, {
      method: 'DELETE'
    })
      .then(resp => {
        if (!resp.ok) {
          throw Error('Error deleting item.')
        }
      })
      .catch(error => console.error(error))
  }

  postNewPurchase = (newPurchase) => {
    const options={
      method: 'POST',
      body: JSON.stringify(newPurchase),
      headers: {
        'Content-Type' : 'application/json'
      }
    }

    fetch('http://localhost:3001/api/v1/purchases', options)
      .then(resp => {
        if (!resp.ok) {
          throw Error('Error posting purchases');
        }
      })
      .catch(error => console.Error(error))
  }

  fetchPurchases = () => {
    return fetch('http://localhost:3001/api/v1/purchases')
      .then(resp => {
        if (!resp.ok) {
          throw Error('Error fetching purchases.');
        }
        return resp.json();
      })
      
  }

  componentDidMount = () => {
    this.fetchPurchases()
    .then(data => this.setState({purchases: data}))
    .catch(error => console.Error('There was an error with the fetch.'))
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1 className='app-title'>My Order History</h1>
          <div className='Form'>
            <Form addNewPurchase={this.addNewPurchase} />
          </div>
        </header>
        <div className='purchase-container'>
          <PurchasesContainer 
          deletePurchase={this.deletePurchase}
          purchases={this.state.purchases} />
        </div>
      </div>
    );
  }
}

export default App;
