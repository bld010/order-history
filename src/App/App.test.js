import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


describe('App', () => {

  let wrapper = null;
  let mockPurchases = [
    {name: 'X'},
    {name: 'Y'},
    {name: 'Z'}
  ]

  beforeEach(()=> {
    wrapper=shallow(<App />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should add new purchases', () => {
    const mockEvent = {
      preventDefault: () => {}
    }
    const mockNewPurchase = {name:'W'}
    const mockPostNewPurchase = jest.fn();
    wrapper.setState({purchases: mockPurchases});

    wrapper.instance().addNewPurchase(mockEvent, mockNewPurchase)
    wrapper.instance().postNewPurchase = mockPostNewPurchase;

    const expected = wrapper.state().purchases

    expect(expected).toEqual([...mockPurchases, mockNewPurchase])
  })

  //postNewPurchase Fetch
  //componentDidMount --> fires fetch (name and move?)
})