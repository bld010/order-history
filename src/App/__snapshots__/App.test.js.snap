// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <header>
    <h1
      className="app-title"
    >
      My Order History
    </h1>
    <div
      className="Form"
    >
      <Form
        addNewPurchase={[Function]}
      />
    </div>
  </header>
  <div
    className="purchase-container"
  >
    <PurchasesContainer
      deletePurchase={[Function]}
      purchases={Array []}
    />
  </div>
</div>
`;
